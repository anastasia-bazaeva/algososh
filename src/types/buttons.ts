export enum ButtonNames {
    Add = "Добавить",
    Remove = "Удалить",
    Clear = "Очистить"
} 