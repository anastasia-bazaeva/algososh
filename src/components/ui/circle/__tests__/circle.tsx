import renderer from 'react-test-renderer';
import { ElementStates } from '../../../../types/element-states';
import { Circle } from '../circle';

const textForCircle = 'A1';
const numberForCircle = 1;
const colors = [
    ElementStates.Default,
    ElementStates.Changing,
    ElementStates.Modified
]

describe('test circle component', () =>{
    it('check render circle with letter', ()=>{
        const tree = renderer.create(<Circle letter={textForCircle}/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render circle without letter', ()=>{
        const tree = renderer.create(<Circle />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render circle with head', ()=>{
        const tree = renderer.create(<Circle head={textForCircle} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render circle with tail', ()=>{
        const tree = renderer.create(<Circle tail={textForCircle} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render circle with node in head', ()=>{
        const tree = renderer.create(<Circle head={<Circle/>} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render circle with node in tail', ()=>{
        const tree = renderer.create(<Circle tail={<Circle/>} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render circle with index', ()=>{
        const tree = renderer.create(<Circle index={numberForCircle} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('check render small circle', ()=>{
        const tree = renderer.create(<Circle isSmall={true}/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    colors.forEach((color => {
        it('check render with color', ()=>{
            const tree = renderer.create(<Circle state={color}/>).toJSON()

            expect(tree).toMatchSnapshot()
        })
    }))
})