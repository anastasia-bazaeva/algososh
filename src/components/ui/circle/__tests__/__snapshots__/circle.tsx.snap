// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle component check render circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    A1
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`test circle component check render circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A1
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render circle with node in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render circle with node in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`test circle component check render circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    A1
  </div>
</div>
`;

exports[`test circle component check render circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render small circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render with color 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render with color 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`test circle component check render with color 3`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
